pkg/apis/seedmanagement/validation/managedseed.go

```diff
@@ -0,0 +1,317 @@
+
+	return allErrs
+}
+
+func validateGardenlet(gardenlet *seedmanagement.Gardenlet, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if gardenlet.Deployment != nil {
+		allErrs = append(allErrs, validateGardenletDeployment(gardenlet.Deployment, fldPath.Child("deployment"))...)
+	}
+
+	if gardenlet.Config != nil {
+		configPath := fldPath.Child("config")
+
+		// Convert gardenlet config to an internal version
+		gardenletConfig, err := helper.ConvertGardenletConfig(gardenlet.Config)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(configPath, gardenlet.Config, fmt.Sprintf("could not convert gardenlet config: %v", err)))
+			return allErrs
+		}
+
+		// Validate gardenlet config
+		allErrs = append(allErrs, validateGardenletConfiguration(gardenletConfig, helper.GetBootstrap(gardenlet.Bootstrap), utils.IsTrue(gardenlet.MergeWithParent), configPath)...)
+	}
+
+	if gardenlet.Bootstrap != nil {
+		validValues := []string{string(seedmanagement.BootstrapServiceAccount), string(seedmanagement.BootstrapToken), string(seedmanagement.BootstrapNone)}
+		if !utils.ValueExists(string(*gardenlet.Bootstrap), validValues) {
+			allErrs = append(allErrs, field.NotSupported(fldPath.Child("bootstrap"), *gardenlet.Bootstrap, validValues))
+		}
+	}
+
+	return allErrs
+}
+
+func validateGardenletUpdate(newGardenlet, oldGardenlet *seedmanagement.Gardenlet, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if newGardenlet.Config != nil && oldGardenlet.Config != nil {
+		configPath := fldPath.Child("config")
+
+		// Convert new gardenlet config to an internal version
+		newGardenletConfig, err := helper.ConvertGardenletConfig(newGardenlet.Config)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(configPath, newGardenlet.Config, fmt.Sprintf("could not convert gardenlet config: %v", err)))
+			return allErrs
+		}
+
+		// Convert old gardenlet config to an internal version
+		oldGardenletConfig, err := helper.ConvertGardenletConfig(oldGardenlet.Config)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(configPath, oldGardenlet.Config, fmt.Sprintf("could not convert gardenlet config: %v", err)))
+			return allErrs
+		}
+
+		// Validate gardenlet config update
+		allErrs = append(allErrs, validateGardenletConfigurationUpdate(newGardenletConfig, oldGardenletConfig, configPath)...)
+	}
+
+	// Ensure bootstrap is not changed
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newGardenlet.Bootstrap, oldGardenlet.Bootstrap, fldPath.Child("bootstrap"))...)
+
+	// Ensure merge with parent is not changed
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(newGardenlet.MergeWithParent, oldGardenlet.MergeWithParent, fldPath.Child("mergeWithParent"))...)
+
+	return allErrs
+}
+
+func validateGardenletDeployment(deployment *seedmanagement.GardenletDeployment, fldPath *field.Path) field.ErrorList {
```

Comment:
Currently, only the image is validated from the GardenletDeployment. What about the other config options like ReplicaCount, RevisionHistoryLimit,... ?

Reply:
Good point, do you have suggestions how they should be validated? I can think of:

* For `ReplicaCount` and `RevisionHistoryLimit` there should be a check for non-negative integer. 
* `PodLabels` and `PodAnnotations` could be validated using the standard functions for validating labels and annotations that are already used elsewhere.

We could also validate other fields such as `Resources`, `Env`, and `AdditionalVolumeMounts`, similarly to how this is done for corev1 containers, see https://github.com/kubernetes/kubernetes/blob/e1ba7546bfe591d4e4fc0ce08c00652b79703b00/pkg/apis/core/validation/validation.go#L2817. However, this code is in `k8s.io/kubernetes`, so my assumption is we don't want to directly depend upon it, so we would need to mostly copy / paste it. Do we really want to do this? It looks like quite a chunk.

Reply:
@rfranzke @timebertt Do you have suggestions here?

Reply:
I would just validate the obvious like ReplicaCount and  RevisionHistoryLimit to not having to dependent or copy from k/k

Reply:
I would probably only implement it for fields which can be validated quite easily (like replica count, revision history limit, etc.). We don't want to depend on `k8s.io/kubernetes` and we don't want to copy all this code. Those fields are anyways "advanced" use-cases, so by default I guess nobody would set them (actually, I don't even know of real-world examples where somebody would need this). Actually, we could also remove them entirely at the moment, nobody would probably set them (it wasn't possible earlier, or?). If you want to keep them then I would live with the fact that failures in configuring those fields properly will result in reconciliation errors for now.

Reply:
OK, let's keep them for now, and I will add the validation for `ReplicaCount` , `RevisionHistoryLimit`, `PodLabels`, and `PodAnnotations`. I will also add a short comment explaining why the rest are not validated.

Reply:
Done